### DISCUSSION TITLE
Filter Rest API using MySQL Table

### DISCUSSION THREAD

**Discussion by smurraystrata** [Budibase/budibase#10156] [Help] [answered]
Hello again!

I have a MySQL table which is updated by automation every day with a list of agent IDs in a specific group. I also have a Rest API call that pulls a list of the status of all agents from every group at the same time. I would like to use the MySQL table to filter the Rest API call to only show those agents in the MySQL table.

I set a filter on the Rest API data source component on the screen to only show those that the agentid matches with the MySQL table, but it only shows the one on the first row of the MySQL table. I set a repeater with the Rest API in it and that fixed the problem but then causes the Rest API to need to be queried for every row on every refresh, which is a bit excessive and causes intermittent performance errors. I tried putting the repeater under the Rest API call and setting the components in the repeater to only show if the Rest API agent ID matches that of the MySQL agent ID, but it didn't have an effect.

Filter applied to Rest API component when placed under the MySQL table component -
![image](https://user-images.githubusercontent.com/120056389/228646030-f9cf9386-da6a-4b1d-af3a-300bc161918b.png)

The condition set on card component inside repeater under Rest API component- 
![image](https://user-images.githubusercontent.com/120056389/228646213-d7045694-d6c3-4e3a-97e5-c9d9e83e4c0b.png)

![image](https://user-images.githubusercontent.com/120056389/228646271-fdb3a9e1-a41a-4ecc-a8ca-cd2c7a497d3a.png)

Any suggestions would be welcome!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @jedman20,

I see you are filtering by whatever the first agent Id is within your rows. Is this agent_id an exact match for all instances? Have you tried outputting the GroupMemberships.Rows.0.agentId to make sure it matches the data you are receiving? 

I can also see you are trying to hide components later on if it also matches, Is there a reason for this? To me this seems like you are hiding the results filtered previously?

Could I ask for an app export in this instance as I'm not quite sure what you are trying to achieve from the screenshots and description you have supplied. Sorry for the trouble!

Thanks!

**Reply by smurraystrata** [reply]
Hello @ConorWebb96,

Sorry for my confusing first post. I would actually like to filter by the agent ID in any row of GroupMemberships. The filter is working for the first row just fine, but I want it to use all the rows in the GroupMemberships table.

I also didn't explain very well that I tried to do this using those two different methods of filtering separately. I was not applying those at the same time. I was just trying to show that I attempted it in different ways and see if there are other ways of possibly filtering using all the rows in GroupMemberships instead of just one, without putting Activity in a repeater (as that caused performance issues).

To explain my end goal in a hopefully better way - Activity reports 50+ agents, I only want the agents to show if their agent ID is in any row of the GroupMemberships table.

I hope that explanation is better, I can break this piece off into a test instance and send it if you are still needing that.

Thank you for your time!

**Reply by smurraystrata** [reply]
Hello again!

I actually just figured this out, I just needed you to point out that in GroupMemberships.Rows.0.agentId = the first row of the table. I'm really not sure why I forgot that. I set conditions on the card inside the repeater listing each row in GroupMemberships and now it is working correctly.

![image](https://user-images.githubusercontent.com/120056389/228851754-fdc51e2d-7b1a-40be-8d30-ecbc7830934d.png)

Thank you greatly!
